src/java/htsjdk/samtools/SAMFileWriterFactory.java

```diff
@@ -293,16 +291,39 @@ public SAMFileWriter makeSAMOrBAMWriter(final SAMFileHeader header, final boolea
         return makeBAMWriter(header, presorted, outputFile);
     }
 
+    /**
+     *
+     * Create a SAM, BAM or CRAM writer based on examination of the outputFile extension.
+     * For CRAM files, read names and tags are preserved.
```

Comment:
Because the CRAM spec allows for a wide variety of lossy and lossless compression strategies, its a bit ambiguous to just say we support CRAM (for example, AFAICT cramtools defaults to not preserving readnames when writing records to a CRAM, and synthesizes readnames on the fly on CRAM decode). This javadoc was an attempt to start making incremental progress on https://github.com/samtools/htsjdk/issues/380. I'm wondering now though if the preservation comment is more confusing than helpful.